### PR TITLE
Improve repository readme

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

# Rationale for this change
 
While working on https://github.com/apache/arrow-rs/pull/5737 and https://github.com/apache/arrow-rs/pull/5750 I noticed a few improvements that could be made to the repository readme.

# What changes are included in this PR?

Refine description and add more links to readme

See the rendered version here: https://github.com/alamb/arrow-rs/blob/alamb/readme/README.md

# Are there any user-facing changes?
Docs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,22 +69,30 @@ For example:
...
 There are two related crates in different repositories
 
-| Crate      | Description                             | Documentation                 |
-| ---------- | --------------------------------------- | ----------------------------- |
-| DataFusion | In-memory query engine with SQL support | [(README)][datafusion-readme] |
-| Ballista   | Distributed query execution             | [(README)][ballista-readme]   |
+| Crate          | Description                             | Documentation                 |
+| -------------- | --------------------------------------- | ----------------------------- |
+| [`datafusion`] | In-memory query engine with SQL support | [(README)][datafusion-readme] |
+| [`ballista`]   | Distributed query execution             | [(README)][ballista-readme]   |
```
Comment (Xuanwo) [inline README.md:75]: Hi, do you think it's a good idea to add `object_store_opendal` here as an integration to `object_store`?

Comment (alamb) [reply README.md:75]: Sounds like a good idea to me -- is it mature enough to steer people to? The crates.io page makes it unclear

https://crates.io/crates/object_store_opendal

Screenshot 2024-05-11 at 6 32 25 AM

Comment (Xuanwo) [reply README.md:75]: > Sounds like a good idea to me -- is it mature enough to steer people to? The crates.io page makes it unclear

Nice catch. I will improve this part and raising a PR later.

Comment (alamb) [reply README.md:75]: Sounds good

**Pull request merged**